### PR TITLE
fix(vite): migrations and adjustments for plugin and config

### PR DESCRIPTION
**Pull request by mandarini**
* remove targetDefaults from vite
* add removed settings inside vite.config.ts

moved to: https://github.com/nrwl/nx/pull/20627 and merged

### CODE REVIEW

File: packages/eslint/src/plugins/plugin.ts
Comment (meeroslav) [inline packages/eslint/src/plugins/plugin.ts:1]: Can you remove these changes as they are already included in the https://github.com/nrwl/nx/pull/20583?

**Review by meeroslav — CHANGES_REQUESTED** [review]
I think the change to Vite's inputs is not correct.

File: packages/vite/src/plugins/plugin.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ async function buildTarget(
     dependsOn: [`^${buildTargetName}`],
     inputs: [
       ...('production' in namedInputs
-        ? ['production', '^production']
+        ? ['default', '^production']
```
[the remark this answers is not in the document]
Comment (meeroslav) [reply packages/vite/src/plugins/plugin.ts:149]: That one might also be incorrect. I see no reason for `build` to depend on non-production files.

**Comment by mandarini** [comment]
@meeroslav  And in cypress it's the same: https://github.com/nrwl/nx/blob/master/packages/cypress/src/plugins/plugin.ts#L292

Maybe I'm not understanding something correctly though, which is quite possible. 😅

**Review by meeroslav — APPROVED** [review]
Thanks for the fixes. LGTM!

**Pull request closed without merging**